From 992932c174d581ad5bf67ee2b2d5fffcdae78198 Mon Sep 17 00:00:00 2001
From: Eike Ziller <eike.ziller@digia.com>
Date: Fri, 6 Dec 2013 12:40:56 +0100
Subject: [PATCH 17/20] OS X Mavericks/Xcode5: Find libxslt and libxml2 headers

The development headers are no longer installed in a system location,
but only in the sysroot.

Change-Id: Ie9b85ff638c435e7e4d0a538f6e72d2cdbae88c8
Reviewed-by: Simon Hausmann <simon.hausmann@digia.com>
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@digia.com>

Conflicts:
	Source/WebCore/WebCore.pri
	Tools/qmake/config.tests/libxml2/libxml2.pro
	Tools/qmake/config.tests/libxslt/libxslt.pro
---
 Source/WebCore/WebCore.pri                   | 2 +-
 Tools/qmake/config.tests/libxml2/libxml2.pro | 9 +++++++--
 Tools/qmake/config.tests/libxslt/libxslt.pro | 9 +++++++--
 3 files changed, 15 insertions(+), 5 deletions(-)

diff --git a/Source/WebCore/WebCore.pri b/Source/WebCore/WebCore.pri
index 9f91080..d671e9c 100644
--- a/Source/WebCore/WebCore.pri
+++ b/Source/WebCore/WebCore.pri
@@ -108,7 +108,7 @@ INCLUDEPATH += $$WEBCORE_GENERATED_SOURCES_DIR
 contains(DEFINES, ENABLE_XSLT=1) {
     contains(DEFINES, WTF_USE_LIBXML2=1) {
         mac {
-            INCLUDEPATH += /usr/include/libxml2
+            QMAKE_CXXFLAGS += -iwithsysroot /usr/include/libxslt -iwithsysroot /usr/include/libxml2
             LIBS += -lxml2 -lxslt
         } else {
             PKGCONFIG += libxslt
diff --git a/Tools/qmake/config.tests/libxml2/libxml2.pro b/Tools/qmake/config.tests/libxml2/libxml2.pro
index d5add8f..7c9baea 100644
--- a/Tools/qmake/config.tests/libxml2/libxml2.pro
+++ b/Tools/qmake/config.tests/libxml2/libxml2.pro
@@ -1,4 +1,9 @@
 CONFIG -= qt
 SOURCES = libxml2.cpp
-CONFIG += link_pkgconfig
-PKGCONFIG += libxml-2.0
+mac {
+    QMAKE_CXXFLAGS += -iwithsysroot /usr/include/libxml2
+    LIBS += -lxml2
+} else {
+    CONFIG += link_pkgconfig
+    PKGCONFIG += libxml-2.0
+}
diff --git a/Tools/qmake/config.tests/libxslt/libxslt.pro b/Tools/qmake/config.tests/libxslt/libxslt.pro
index 6359de3..967fe16 100644
--- a/Tools/qmake/config.tests/libxslt/libxslt.pro
+++ b/Tools/qmake/config.tests/libxslt/libxslt.pro
@@ -1,4 +1,9 @@
 CONFIG -= qt
 SOURCES = libxslt.cpp
-CONFIG += link_pkgconfig
-PKGCONFIG += libxslt
+mac {
+    QMAKE_CXXFLAGS += -iwithsysroot /usr/include/libxslt -iwithsysroot /usr/include/libxml2
+    LIBS += -lxslt
+} else {
+    CONFIG += link_pkgconfig
+    PKGCONFIG += libxslt
+}
-- 
1.8.5.3

